diff --git a/lib/webrtc/src/transport/ice.js b/lib/webrtc/src/transport/ice.js
index 25c8489..cd06dbf 100644
--- a/lib/webrtc/src/transport/ice.js
+++ b/lib/webrtc/src/transport/ice.js
@@ -230,25 +230,25 @@ class RTCIceTransport {
             stats.push(candidateStats);
         }
         // Candidate pairs
         const pairs = this.connection?.candidatePairs
             ? [
                 ...this.connection.candidatePairs.filter((p) => p.nominated),
                 ...this.connection.candidatePairs.filter((p) => !p.nominated),
             ]
             : [];
         for (const pair of pairs) {
             const pairStats = {
                 type: "candidate-pair",
-                id: (0, stats_1.generateStatsId)("candidate-pair", pair.foundation),
+                id: (0, stats_1.generateStatsId)("candidate-pair", pair.localCandidate.foundation, pair.remoteCandidate.foundation),
                 timestamp,
                 transportId: (0, stats_1.generateStatsId)("transport", this.id),
                 localCandidateId: (0, stats_1.generateStatsId)("local-candidate", pair.localCandidate.foundation),
                 remoteCandidateId: (0, stats_1.generateStatsId)("remote-candidate", pair.remoteCandidate.foundation),
                 state: pair.state,
                 nominated: pair.nominated,
                 packetsSent: pair.packetsSent,
                 packetsReceived: pair.packetsReceived,
                 bytesSent: pair.bytesSent,
                 bytesReceived: pair.bytesReceived,
                 currentRoundTripTime: pair.rtt,
             };
